src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubUpload.java

```diff
@@ -50,23 +66,57 @@ public void execute() throws MojoExecutionException {
                 + ", version: " + version
                 + ", inputFile: " + inputFile
                 + ", format: " + format
-                + ", isPrivate: " + isPrivate);
+                + ", isPrivate: " + isPrivate
+                + ", definitionDirectory: " + definitionDirectory
+                + ", definitionFileNameRegex: " + definitionFileNameRegex
+                + ", uploadType: " + uploadType);
+
+        Optional<DefinitionUploadType> definitionUploadType = DefinitionUploadType.getByParamValue(uploadType);
+        definitionUploadType.orElseThrow(() -> new MojoExecutionException(String.format("Unknown uploadType [%s] specified. Supported types are inputFile and directory.", uploadType)));
 
         try {
-            String content = new String(Files.readAllBytes(Paths.get(inputFile)), Charset.forName("UTF-8"));
-            String oasVersion = DefinitionParserService.getOASVersion( DefinitionFileFormat.valueOf(format.toUpperCase()).getMapper().readTree(content));
+            if(definitionUploadType.get().equals(DefinitionUploadType.INPUT_FILE)) {
+                String content = new String(Files.readAllBytes(Paths.get(inputFile)), Charset.forName(UTF_8));
```

Comment:
why not just
`SwaggerHubRequest swaggerHubRequest = createSwaggerHubRequest(file);
swaggerHubClient.saveDefinition(swaggerHubRequest);`